apps/app.go

```diff
@@ -45,23 +44,20 @@ func (id AppId) String() string {
 	return string(id)
 }
 
-func (id AppId) ConsulServiceName() string {
-	return strings.Replace(strings.Trim(strings.TrimSpace(id.String()), "/"), "/", ".", -1)
-}
-
 func (app *App) IsConsulApp() bool {
 	_, ok := app.Labels[MARATHON_CONSUL_LABEL]
 	return ok
 }
 
-func (app *App) ConsulServiceName() string {
-	if value, ok := app.Labels[MARATHON_CONSUL_LABEL]; ok && value != "true" {
-		value = AppId(value).ConsulServiceName()
-		if value != "" {
-			return value
-		}
+func (app *App) ConsulName() string {
```

Comment:
In my opinion it should stay here as it's related to the Marathon app configuration. Here `ConsulName` returns the name that the user set for the app on the Marathon side (using it's labels or simply getting the default app-id). That's why we do the `"true"` and `""` comparison here, etc. 
On the Consul class side we have the destination service name returned with all the Consul logic applied.